### PR TITLE
WIP:Documentation/citation style

### PR DESCRIPTION
**Pull request by PhRosenberger**
#### Reference to a related issue in the repository
Related to #356 

#### Add a description
This is my proposal for a citation style. Please check before I change all existing citations and add missing ones.

#### Mention a member
@jdsika could you do the check?

#### Check the checklist

- [x] My code and comments follow the [style guidelines](https://opensimulationinterface.github.io/osi-documentation/osi/commenting.html) and [contributors guidelines](https://opensimulationinterface.github.io/osi-documentation/osi/howtocontribute.html) of this project.
- [x] I have performed a self-review of my own code.
- [x] I have made corresponding changes to the [documentation](https://github.com/OpenSimulationInterface/osi-documentation).
- [x] My changes generate no new warnings.
- [ ] I have added tests that prove my fix is effective or that my feature works.
- [ ] New and existing unit tests / travis ci pass locally with my changes.

### CODE REVIEW

**Comment by jdsika** [comment]
What citation style is this?

File: doc/commenting.rst [hunk 1]
```diff
@@ -199,7 +198,26 @@ You can optionally add a note to the field to describe the field better.
...
+Every OSI message should be defined properly and should have a well cited reference.
+
+**Citation style for different sources:**
+
+- Find filled-out examples in already existing entries
+- All citations should be primary citations. Sources like Wikipedia et al. are not allowed.
+- For more than 3 authors, place "et al." after the third one
+- For more than 1 editor, place "et al." after the first one
+- Naming pages at the end is optional, to ease finding in long texts or for direct citations
+- Scheme is as follows (replace tags with corresponding values):
```
Comment (jdsika) [inline doc/commenting.rst]: agree

**Comment by PhRosenberger** [comment]
> What citation style is this?

It is similiar to APA, but not identical. Here, the <year> is at the end, whereas APA defines the <year> in () after the author.

We can go for APA completely to have something standardized.
What do you think?

In the meantime, I continue and try out italic in document title of the reference, so there will be some slight changes.

**Comment by jdsika** [comment]
Complete apa is good.

**Comment by PhRosenberger** [comment]
So we will go for full APA style.

**Comment by PhRosenberger** [comment]
I have pushed the citation style section in the doc.
Next step is to change existing citations and add missing ones.

File: doc/commenting.rst [hunk 2]
```diff
@@ -199,7 +198,31 @@ You can optionally add a note to the field to describe the field better.
...
     }
 
-If you want to provide a reference to a DIN or to web page which helps in understanding the field you can add a reference.
+To help understanding the field, you should add a reference.
+Every OSI message should be defined properly and should have a well cited reference.
+
+**Citation style for different sources:**
+
+- Within the text, the number system is used with the number of the source in brackets [#] for mentioning.
+- We use the so called "APA style"`link <https://apastyle.apa.org/>`_ from the American Psychological Association for referencing ().
```
Comment (vkresch) [inline doc/commenting.rst]: Change the link into "`APA style <https://apastyle.apa.org/>`_" and remove the brackets at the end.

**Comment by vkresch** [comment]
@jdsika @PhRosenberger Just as side note for checking the sphinx rst syntax I use [this](https://livesphinx.herokuapp.com/) site.

File: doc/commenting.rst [hunk 3]
```diff
@@ -199,7 +198,31 @@ You can optionally add a note to the field to describe the field better.
...
+- In the references list, the number in brackets [#] is followed by a full citation.
+- For writing the title in italic, use <em>title</em>
+- Author names are written as <surname>, <initial(s)> like Authorname, A. A.
+- Editor names are written as <initial(s)> <surname> like B. B. Editorname
```
Comment (vkresch) [inline doc/commenting.rst]: (Minor issue) Decide on consistent period usage for bullet points.

File: osi_environment.proto [hunk 4]
```diff
@@ -129,7 +129,7 @@ message EnvironmentalConditions
     // contrast of a distant object against its background.
     //
     // \par References:
-    // - [2] SHEPARD, Frank D. Reduced visibility due to fog on the highway.
+    // - [2] SHEPARD, Frank D. <em>Reduced visibility due to fog on the highway.</em>
```
Comment (vkresch) [inline osi_environment.proto]: Should the bullet point here be removed too like in the example in comment.rst?

File: osi_environment.proto [hunk 5]
```diff
@@ -129,7 +129,7 @@ message EnvironmentalConditions
...
     // Transportation Research Board, 1996.
     // - [3] [StVO 17 chapter
```
Comment (vkresch) [inline osi_environment.proto]: Here, too? (bullet point removal)

File: osi_environment.proto (hunk 4 above)
Comment (PhRosenberger) [reply osi_environment.proto]: I am not done with the  .proto files, so yes, I will remove the bullet points.

File: doc/commenting.rst (hunk 2 above)
Comment (PhRosenberger) [reply doc/commenting.rst]: Thank you for the support, will be changed.

File: doc/commenting.rst (hunk 3 above)
Comment (PhRosenberger) [reply doc/commenting.rst]: Yes.

File: osi_environment.proto (hunk 5 above)
Comment (PhRosenberger) [reply osi_environment.proto]: Yes.

File: doc/commenting.rst [hunk 6]
```diff
@@ -243,7 +243,7 @@ Every OSI message should be defined properly and should have a well cited refere
         // \note 100000 Pa = 1 bar
         //
         // \par Reference:
-        // [1] Rapp, C.: Grundlagen der Physik, in: Hydraulik für Ingenieure und Naturwissenschaftler, Springer Vieweg, Wiesbaden, 2017, p. 105
+        // [1] Rapp, C. (2017). Grundlagen der Physik. <em>In Hydraulik für Ingenieure und Naturwissenschaftler</em> (pp.23-36). Springer Vieweg. Wiesbaden, Germany. https://doi.org/10.1007/978-3-658-18619-7_3. p. 105.
```
Comment (PhRosenberger) [inline doc/commenting.rst]: Actually, I am wondering if the length of the line for the reference in the comments could be to long like this.
@vkresch and @jdsika is there a limitation or style-standard for this?

**Comment by PhRosenberger** [comment]
So, I am done with the existing citations. For adding new citations, where they are missing, I would propose to create a new branch. Actually, it took a lot of time to find a good citation style and especially for correcting the existing ones, so finding missing citations will take probably even more time.

A possible work-around could be to wait for ISO23150 to be publicly available and cite this then. What do you think, @jdsika, @vkresch, and others?

**Comment by PhRosenberger** [comment]
Actually, the line breaking with " \n " from the last commit fails because of the non-ASCII testing.
Should we change the code or the test?

**Comment by vkresch** [comment]
@PhRosenberger it's mostly because of the character "ß" in "Straßenverkehrs-Ordnung" and "ü" in "für". So basically we need to decide if we want to cite **only** in english or  **only** in german. If we decide on german then we need to remove the non-ascii test. This could cause some bugs/erros in the doxygen generation since this is the reason behind the test. This should be tested accordingly.

**Comment by jdsika** [comment]
I would go for an "english only" style

**Comment by PhRosenberger** [comment]
> I would go for an "english only" style

Does this mean only English sources?
Could be difficult regarding e.g. DIN standards..

I prefer to change writing into "ss", "ue", "ae", and "oe".

**Comment by PhRosenberger** [comment]
Ready for merge into master now.

**Comment by PhRosenberger** [comment]
@jdsika and @vkresch can we merge this branch now?

**Comment by jdsika** [comment]
I am fine with ö as oe!
@vkresch please review

**Comment by jdsika** [comment]
@PhRosenberger thank you very much - good job!

**Pull request merged**